dfastmi/batch/core.py

```diff
@@ -571,102 +580,138 @@ def analyse_and_report(
-    kmbounds : Tuple[float,float]
-        Minimum and maximum chainage values indicating range of interest.
+    rootdir : str
+        Reference directory for default folders.        
     outputdir : str
-        Name of the output directory.
-    plotops : Dict
-        Dictionary of plot settings
+        Reference directory for default output folders.
+    gui : bool
+        Flag indicating whether this routine is called from the GUI.
 
     Returns
     -------
     success : bool
         Flag indicating whether analysis could be carried out.
     """
+    initialized_config = ConfigurationInitializerFactory.generate(cfg_version, reach, config)
+    
+    old_zmin_zmax = False
+
+    # set plotting flags
+    plotops = _set_plotting_flags(rootdir, display, data)
+
+    imode = _log_report_mode_usage(config, report)
+    filenames = get_filenames(imode, initialized_config.needs_tide, config)
+    success = False
+
     if imode == 0:
-        return AnalyserAndReporterWaqua.analyse_and_report_waqua(
+        success = AnalyserAndReporterWaqua.analyse_and_report_waqua(
             display,
             report,
             reduced_output,
-            tstag,
-            Q,
-            apply_q,
-            T,
-            rsigma,
-            ucrit,
+            initialized_config.tstag,
+            initialized_config.discharges,
+            initialized_config.apply_q,
+            initialized_config.time_fractions_of_the_year,
+            initialized_config.rsigma,
+            initialized_config.ucrit,
             old_zmin_zmax,
             outputdir,
         )
     else:
-        return AnalyserAndReporterDflowfm.analyse_and_report_dflowfm(
+        success = AnalyserAndReporterDflowfm.analyse_and_report_dflowfm(
             display,
             report,
             reach,
-            q_location,
-            q_threshold,
-            tstag,
-            Q,
-            apply_q,
-            T,
-            rsigma,
-            slength,
-            nwidth,
-            ucrit,
+            branch.qlocation,
+            initialized_config.q_threshold,
+            initialized_config.tstag,
+            initialized_config.discharges,
+            initialized_config.apply_q,
+            initialized_config.time_fractions_of_the_year,
+            initialized_config.rsigma,
+            initialized_config.slength,
+            reach.normal_width,
+            initialized_config.ucrit,
             filenames,
-            xykm,
-            needs_tide,
-            n_fields,
-            tide_bc,
+            plotops['xykm'],
+            initialized_config.needs_tide,
+            initialized_config.n_fields,
+            initialized_config.tide_bc,
             old_zmin_zmax,
-            kmbounds,
+            plotops['kmbounds'],
             outputdir,
             plotops,
         )
 
+    _log_length_estimate(report, initialized_config.slength)
```

Comment:
maybe move the log and plot related parts into seperate classes or modules?

Reply:
not needed